src/bsp/src/stm32/src/InterlocManager.cpp

```diff
@@ -66,3 +82,174 @@ bool InterlocManager::constructUWBHeader(uint16_t destinationId,
 
     return true;
 }
+
+void InterlocManager::setCalibDistance(uint16_t distanceForCalibCm) {
+    m_distanceForCalibCm = distanceForCalibCm;
+}
+
+void InterlocManager::startCalibSingleInit() {
+    // TODO add destination ID
+    m_decaA.setState(DW_STATE::SEND_CALIB);
+}
+
+void InterlocManager::startCalibSingleRespond() {
+    // TODO add destination ID
+    if (m_decaA.getState() != DW_STATE::CALIBRATED) {
+        m_decaA.setState(DW_STATE::RESPOND_CALIB);
+    } else if (m_decaB.getState() != DW_STATE::CALIBRATED) {
+        m_decaA.setState(DW_STATE::RESPOND_CALIB);
+    }
+}
+
+void InterlocManager::startCalibSingleInit(uint16_t destinationId, Decawave& device) {
+    device.setChannel(UWBChannel::CHANNEL_2);
+    device.setTxAntennaDLY(DEFAULT_TX_ANT_DLY);
+    device.setRxAntennaDLY(DEFAULT_RX_ANT_DLY);
+
+    while (device.getState() == DW_STATE::SEND_CALIB) { // find exit condition
+        sendTWRSequence(destinationId, device);
+        Task::delay(100);
+    }
+}
+
+void InterlocManager::startCalibSingleRespond(uint16_t destinationId, Decawave& device) {
+    int32_t error;
+    int32_t dwCountOffset;
+    // distance between devices in calibration mode in centimeters
+
+    device.setChannel(UWBChannel::CHANNEL_2);
```

Comment:
should the channel be hardcoded? do we want a compilation setting or at least a define?

Reply:
created DEFAULT_CHANNEL is Decawave.h